daphne/src/lib.rs

```diff
@@ -48,6 +48,10 @@ pub enum DapError {
     #[error("fatal error: {0}")]
     Fatal(String),
 
+    /// Bad request error. If this triggers an abort, then treat this as an external error.
+    #[error("bad request error: {0}")]
+    BadRequest(String),
```

Comment:
I think an explicit error would be best here. Eventually we'll need to address this case in the spec itself.

```suggestion
    /// Received an unexpected message.
    // 
    // TODO spec: Handle this edge case in the spec itself.
    #[error("bad request error: {0}")]
    UnexpectedMessage,
```

Reply:
We have removed `BadRequest` entirely.